File: Triangle.java
package triangle;
import java.util.*;

public class Triangle {
	static Scanner scn = new Scanner(System.in);
	String answer;
	Triangle(){
		System.out.println("Hello puts please the sides of a traingle, like: 12,21,12");
		answer = scn.nextLine();
		while(notOk(answer)){
			System.out.println("Your line contains wrong symbols, please try again, like: 12,21,12");
			answer = scn.nextLine();
		}
	}	
	
	public boolean daesItTriang(String  answer){
		String[] arrOfSides = answer.split(",");
		int a = Integer.parseInt(arrOfSides[0]);
		int b = Integer.parseInt(arrOfSides[1]);
		int c = Integer.parseInt(arrOfSides[2]);
		if(a >= b+c || b >= a+c || c >= b+a ){
			return false;
		}else{
			return true;
		}	
	}
	public void whichOne(String  answer){
		String[] arrOfSides = answer.split(",");
		int a = Integer.parseInt(arrOfSides[0]);
		int b = Integer.parseInt(arrOfSides[1]);
		int c = Integer.parseInt(arrOfSides[2]);
		if(a == b || c == b || a == c ){
			System.out.println("It's isosceles triangle");
		}
		if(a == b && c == b){
			System.out.println("It's equilateral triangle");
		}
		if(Math.pow(a,2) == Math.pow(b,2) + Math.pow(c,2) || Math.pow(b,2) == Math.pow(a,2) + Math.pow(c,2) || Math.pow(c,2) == Math.pow(b,2) + Math.pow(a,2)){
			System.out.println("It's rectangular triangle");
		}	
	}
	private static boolean notOk(String  answer){
		if(answer != null ){
			String[] arr = answer.split(",");
			if(answer != null && answer.matches("[0-9,]+") && arr.length == 3){
				return false;
				}else{
				return true;
			}
		}else{
			return true;
		}	
		
		
	}
	public static void main(String[] args) {
		Triangle tr = new Triangle();
			if(tr.daesItTriang(tr.answer)){
				System.out.println("It's a triangle!");
				tr.whichOne(tr.answer);
			}else{
				System.out.println("It's not a triangle, sorry!");
			}
	}
}

File: triangle.rb
class Triangle
	def initialize
		puts 'Hello puts sides of traingle, like: 12,21,12'
		@answer = gets.chomp
		while !its_ok?		
		puts 'Wrong syblos is exist in the string, please repiat:'
			@answer = gets.chomp
		end
	end

	def its_triang? 
		arr = @answer.split(',')
		if arr[0].to_i >= arr[1].to_i + arr[2].to_i || arr[2].to_i >= arr[1].to_i + arr[0].to_i || arr[1].to_i >= arr[2].to_i + arr[0].to_i 
		then return false
		else return true
		end
	end
	
	def whats_kind
		arr = @answer.split(',')
		a = arr[0].to_i
		b = arr[1].to_i
		c = arr[2].to_i 
		if a == b && b == c
		then puts 'Its equilateral triangle'
		end
		if a == b || b == c || a == c
		then puts 'Its isosceles triangle'
		end
		if a**2 == b**2 + c**2 || b**2 == a** + c**2 || c**2 == b**2 + a**2
		then puts 'Its rectangular triangle'
		end
	end	
	
private
	def its_ok? 
		if	@answer.scan(/[^0-9,]/).size == 0 && @answer.split(',').size == 3
		then return true
		else return false
		end
	end 
end
tr = Triangle.new
if tr.its_triang?
	then tr.whats_kind		
	else puts 'Its not a triangle, sorry!'
end
